pkg/controller/provider/container/vsphere/collector.go

```diff
@@ -508,18 +511,21 @@ func (r *Collector) connect(ctx context.Context) error {
 	soapClient.SetThumbprint(url.Host, r.thumbprint())
 	vimClient, err := vim25.NewClient(ctx, soapClient)
 	if err != nil {
-		return liberr.Wrap(err)
+		return
 	}
 	r.client = &govmomi.Client{
 		SessionManager: session.NewManager(vimClient),
 		Client:         vimClient,
 	}
 	err = r.client.Login(ctx, url.User)
 	if err != nil {
-		return liberr.Wrap(err)
+		if strings.Contains(err.Error(), "Cannot complete login due to an incorrect user name or password") {
```

Comment:
Really don't like this solution but the only alternative I can think of is to rewrite the communication for VMware to basic http requests. Which would need to include the requests for the ca token to validate the thumbprint. Not sure if it would be worth this issue.
(I read docs for the govmomi and could not find any function which could return the http status code of url.)

Reply:
lgtm